pkg/channeld/data.go

```diff
@@ -145,11 +148,12 @@ func (d *ChannelData) OnUpdate(updateMsg common.ChannelDataMessage, t ChannelTim
 	} else {
 		mergeWithOptions(d.msg, updateMsg, d.mergeOptions, spatialNotifier)
 	}
-
+	d.msgIndex = d.msgIndex + 1

```

Comment:
Should we take the integer overflow into account?

Reply:
if we don't mind the memory cost of using uint64 as message index, we can consider this will never trigger an overflow.

Reply:
Sorry for the misunderstanding - I mean, should we consider the situation that msgIndex can go beyond the limit of uint64? If there are 100 update messages sent to a channel per second, the overflow will take many years to happen, which is fine by me :)